### PR TITLE
[FrameworkBundle] [SecurityBundle] Rename internal WebTestCase to avoid confusion

### PR DESCRIPTION
**Pull request by janvt**
FrameworkBundle & SecurityBundle had 2 classes called WebTestCase, 
one of which is only meant for internal tests. To avoid confusion the internal
classes have been renamed to AbstractWebTestCase and made abstract.

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no (or, yes, but internal class)
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32577
| License       | MIT

### CODE REVIEW

**Comment by janvt** [comment]
This change follows the instructions in the issue, however, I will say it feels wierd that the `WebTestCase` class is abstract and is inherited by the `AbstractWebTestCase`, which is not abstract (but could be)....

**Comment by nicolas-grekas** [comment]
Shouldn't this change target 3.4?

**Comment by janvt** [comment]
Wasn't sure, can gladly reopen a new PR.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/AbstractWebTestCase.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase as BaseWebTestCase;
 use Symfony\Component\Filesystem\Filesystem;
 
-class WebTestCase extends BaseWebTestCase
+class AbstractWebTestCase extends BaseWebTestCase
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/AbstractWebTestCase.php]: :+1: for `abstract class`

**Comment by janvt** [comment]
So, if I change this to target the 3.4 branch, the patch will not cleanly apply to versions after, as some functional tests were only added in later versions. How should I proceed?

**Comment by xabbuh** [comment]
@janvt We can either handle that in follow-up PRs or someone who merges branches up can fix those cases while merging.

**Comment by janvt** [comment]
So, should I just create one PR for each branch (`3.4`& `4.*`)? Can gladly do that, just a bit of busy work :)

**Comment by janvt** [comment]
Also just saw, the same is the case in the `SecurityBundle` - `\Symfony\Bundle\SecurityBundle\Tests\Functional\WebTestCase`. Currently working on a PR, should I rename this as well? Seems to be internal, same as the `FrameworkBundle` class.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/AbstractWebTestCase.php (hunk 1 above)
Comment (janvt) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/AbstractWebTestCase.php]: Done.

**Comment by janvt** [comment]
I opened 2 PRs against the `3.4` and `4.2` branches. The latter should patch relatively well onto `4.*` branches. only one test removed, as far as I can see. Will close this.

**Pull request closed without merging**